### PR TITLE
feat(axisPointer): add option to disable emphasis. close #18495

### LINKED ISSUE #18495: [Feature] Add ability to disable emphasis/highlight functionality of tooltip axis trigger
### What problem does this feature solve?

It would be really helpful to be able to render the tooltip axis trigger (e.g. a line or a cross) while disabling the highlight/emphasis functionality it triggers on the underlying series. This would probably be a configuration option for `tooltip` or `tooltip.axisPointer`.

I created a very simple example where you can see the behavior based on one of the line chart examples.
https://codesandbox.io/s/line-chart-tooltip-highlight-issue-ob1ob0?file=/index.js

When you have a line chart with multiple series, the axis trigger tends to highlight all the series, which is often unhelpful and can be really confusing to users.
image

If you focus very specifically on a line in this use case, you get a single focus. This behavior is fine.
image

If you set `tooltip.trigger` to `none`, you get more helpful highlight behavior where nothing is highlighted when you're outside a series, but you lose the visual assistance of the axis pointer tracking line to orient yourself within the chart.
image

A single series is highlighted when you are hovered over it, so that behavior is the same.
image

I tried a few different combinations of series, tooltip, and axis pointer configuration options to see if I could generate what I wanted with the current functionality, but I could not find a way. It seems like it's currently possible to disable highlight behavior entirely and disable the tooltip axis pointer. I'm hoping for the ability to have **only** series-level highlight functionality while rendering an axis pointer.

### What does the proposed API look like?

This would probably be a configuration option for `tooltip` and/or `tooltip.axisPointer`. I do not have strong feelings about the name of the option to trigger this behavior. I see a `silent` option on some other parts of the API, so maybe having `tooltip.axisPointer.silent: true` could trigger "render but do not highlight" behavior, but I'm not sure if that would break other tooltip functionality (e.g. showing the tooltip on hover). Maybe a more specific `emphasisTrigger` or something like that would be more helpful. I'm open to whatever API approach folks with more experience with the internals of the library feel is best.

It looks like the implementation change would probably go hereabouts in the code. https://github.com/apache/echarts/blob/f5bb3b0c81f9843f73f57a0b01a35488f7795dd6/src/component/axisPointer/axisTrigger.ts#L447-L495

### PR DESCRIPTION
**Pull request by juliepagano**
## Brief Information

This pull request is in the type of:

- [ ] bug fixing
- [X] new feature
- [ ] others

### What does this PR do?

This PR adds a `triggerEmphasis` option to the `axisPointer` that enables the user to disable the emphasis/highlight behavior associated with the axis pointer.

### Fixed issues

- #18495

### Before: What was the problem?

Prior to this PR, the axis pointer would always trigger highlight behavior. When you have a line chart with multiple series, the axis trigger tends to highlight all the series, which is often unhelpful and can be really confusing to users.

image

There is not a good way to work around this currently. You can see an example of some issues related to this in https://github.com/perses/perses/pull/1112.

### After: How does it behave after the fixing?

After this PR, the user has the option to disable this behavior on a given axis by setting `axisPointer.triggerEmphasis` to `false`. When set to `false`, the highlight/emphasis behavior will not trigger with the axis pointer. Other highlight/emphasis (e.g. on mouseover a specific line) is retained.

The default value for `axisPointer.triggerEmphasis` is `true` when not set, retaining the prior behavior.

Below is a screenshot of the `axisPointer.triggerEmphasis: false` visual test. You can see the diff is whether or not all the lines are highlighted. Note that it still retains highlight behavior on mousing over a particular line.
image

## Document Info

One of the following should be checked.

- [ ] This PR doesn't relate to document changes
- [ ] The document should be updated later
- [X] The document changes have been made in https://github.com/apache/echarts-doc/pull/343

~I'm happy to take a stab at making a PR to the docs once we're sure about the naming of the new option. I may need some help with the zh-based language bits. Let me know if you'd prefer I help with this or leave it to the maintainers.~ One of my teammates helped me with the translation, but it could use a second set of eyes to make sure we got it right.

### ZRender Changes

- [ ] This PR depends on ZRender changes (ecomfe/zrender#xxx).

### Related test cases or examples to use the new APIs

I added two test cases to `test/tooltip-axisPointer2.html`. One for the `false` case and one for the `true` case. Let me know if you think I should break this out into a separate test case, and if so, what you think the best name is for it.

I added visual tests for each test case that take screenshots at the following points:
- When mousing over a section of the chart without a line. This validates the axis pointer's configuration is behaving as expected.
- When mousing over an individual line. This validates non-axis-pointer highlight behavior is not impacted.

### Merging options

- [x] Please squash the commits into a single one when merging.

### Other information

I was unsure of the best name for this configuration option. This behavior is referred to as `emphasis` in the configuration of the styling, but is referred to as `highlight` when triggering the event. I started with `triggerEmphasis` because I thought users might be more familiar with the styling configuration, but I can see arguments for `triggerHighlight`. I used the `trigger*` naming to match the similar-ish `triggerTooltip` option. I'm happy to rename it to whatever the maintainers think is best.

### CODE REVIEW

**Review by Ovilia — CHANGES_REQUESTED** [review]
The code looks good to me but it seems some of the tooltip-related test cases fail in the visual test.

image

@juliepagano Can you have a look?

**Comment by juliepagano** [comment]
> The code looks good to me but it seems some of the tooltip-related test cases fail in the visual test.
> 
> image
> 
> @juliepagano Can you have a look?

Sure thing! The diff from "tooltip-axisPointer2" is from the case where `triggerEmphasis` is `false`, so that's expected (screenshot below). I'll take a closer look at the other failing tests to see what's going on there. 

image

File: src/component/axisPointer/modelHelper.ts [hunk 1]
```diff
@@ -194,8 +196,9 @@ function collectAxesInfo(result: CollectionResult, ecModel: GlobalModel, api: Ex
                 : axisPointerModel;
 
             const snap = axisPointerModel.get('snap');
+            const triggerEmphasis = axisPointerModel.get('triggerEmphasis');
             const axisKey = makeKey(axis.model);
-            const involveSeries = triggerTooltip || snap || axis.type === 'category';
+            const involveSeries = triggerTooltip || triggerEmphasis || snap || axis.type === 'category';
```
Comment (juliepagano) [inline src/component/axisPointer/modelHelper.ts:201]: I _think_ the addition of `triggerEmphasis` here might be causing the unexpected visual diffs. Removing it and testing it out locally to validate.

**Comment by juliepagano** [comment]
The visual tests look much better after the commit I made to remove `triggerEmphasis` from the conditional for `involveSeries`. The only failing tooltip test is now the expected one for `triggerEmphasis: false`.

Screenshot of the expected visual diff and showing all the other tooltip tests passing.
image

Scrolling down to screenshot the other tests.
image

**Comment by juliepagano** [comment]
I'm working with one of my teammates to get help with the Chinese translation, so I can put up a docs PR to go with this. Hoping I'll have that up some time today.

**Review by Ovilia — APPROVED** [review]

**Comment by Ovilia** [comment]
Thank you for this high-quality pull request!

**Comment by Ovilia** [comment]
We ususally release a new version every two months. If you want to use this feature ASAP, you can use the [nightly](npm install echarts@npm:echarts-nightly) version in your development environment.

**Pull request merged**